app/code/ZP/LoyaltyProgram/Console/Command/Customer/AssignLoyaltyProgram.php

```diff
@@ -0,0 +1,200 @@
+<?php
+declare(strict_types=1);
+
+namespace ZP\LoyaltyProgram\Console\Command\Customer;
+
+use Magento\Customer\Api\CustomerRepositoryInterface;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Customer\Api\Data\CustomerInterface;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use ZP\LoyaltyProgram\Api\LoyaltyProgramManagementInterface;
+use ZP\LoyaltyProgram\Model\Prepare\Data\DataPreparer;
+use ZP\LoyaltyProgram\Model\Validators\Data\Validator;
+use ZP\LoyaltyProgram\Model\MessageManager\Console\Command\Customer\AssignLoyaltyProgram\MessageManager;
+
+class AssignLoyaltyProgram extends Command
+{
+    private const CUSTOMER_IDS = 'customer_ids';
+    private bool $isAllCustomers;
+
+    /**
+     * @param string|null $name
+     */
+    public function __construct(
+        private CustomerRepositoryInterface $customerRepository,
+        private SearchCriteriaBuilder $searchCriteriaBuilder,
+        private LoyaltyProgramManagementInterface $loyaltyProgramManagement,
+        private DataPreparer $prepareData,
+        private Validator $dataValidator,
+        private MessageManager $messageManager,
+        string $name = null
+    ) {
+        parent::__construct($name);
+    }
+
+    protected function configure(): void
+    {
+        $this->setDescription(
+            'A command to assign loyalty program to customer(s).'
+        );
+
+        $this->addArgument(
+            self::CUSTOMER_IDS,
+            InputArgument::IS_ARRAY,
+            'Separate multiple ids with a space.'
+        );
+
+        parent::configure();
+    }
+
+
+    public function execute(InputInterface $input, OutputInterface $output)
+    {
+        try {
+            $customerIds = $this->validateCustomerIds($this->getCustomerIds($input));
+            if (
+                !$this->isAllCustomers() && !$customerIds &&
+                !$this->messageManager->isResultTypeEmpty(MessageManager::WRONG_DATA)
+            ) {
+                $output->writeln($this->messageManager->getResultMessage(MessageManager::WRONG_DATA));
+
+                return 1;
+            }
+
+            /** @var CustomerInterface[] $customers */
+            $customers = $this->getCustomers($customerIds);
+            if (!$this->compareCustomersCount($customers, $customerIds) && !$customers) {
+                if ($this->messageManager->isResultTypeEmpty(MessageManager::NOT_EXIST)) {
+                    $output->writeln('We don\'t have any customers at the moment!');
+
+                    return 1;
```

Comment:
please check `\Magento\Framework\Console\Cli::RETURN_SUCCESS `
and `\Magento\Framework\Console\Cli::RETURN_FAILURE`